### PR TITLE
Reader: search on manage, 'show more'

### PR DESCRIPTION
**Pull request by samouri**
Addition of the show more button + associated functionality:
1. default state = recs + subs
2. search active = 3 results + button + subs below
3. clicked button = just search results (same as currently doing a search)

cc @jancavan 

manage-following- desktop 003

### CODE REVIEW

**Comment by samouri** [comment]
I've noticed a bug that occurs during the state where a search occurred but there show more wasn't clicked yet: rv is messing up the heights and strange clipping occurs while scrolling through subs.

**Comment by jancavan** [comment]
I've added styles to the "more" button in: https://github.com/Automattic/wp-calypso/pull/13480/commits/30bb1356a1275a4fbab5bd90c2367c45820aeecc

**Comment by samouri** [comment]
Things I noticed that need to be fixed:
- investigate `scrollToTop` issue.  When scrolling quickly through a list with items that haven't loaded yet...for some reason...it jumps you to the top.  Looks very similar to this issue: https://github.com/bvaughn/react-virtualized/issues/595.  need to see if the scrollToTop is happening in our code or is a regression in rv.
- clipping: I've noticed that when the devtools are open and the search is in its half-expanded state, i can sometimes repro strange clipping.
- loading placeholders: we need them.   Could solve the scrollToTop issue if we have something to show in the meantime

**Comment by samouri** [comment]
another thing:
- should isExpanded be part of the url? Ideally the back button would also bring you to an expanded state after clicking into a feed.  @blowery?

**Comment by samouri** [comment]
I've created a plunker where I can repro the scrollToTop issue without even introducing the InfiniteLoader: https://plnkr.co/edit/gj1zaNt8yQXfIH6ahMpU?p=preview

**Comment by blowery** [comment]
> should isExpanded be part of the url? Ideally the back button would also bring you to an expanded state after clicking into a feed. @blowery?

Hrm. Since we hide the following list on expand, probably. Maybe it should be `/read/following/manage/search?q=blah` ?

**Comment by blowery** [comment]
or if it's easier, yeah, just add in an `expanded=true` querystring argument

**Comment by samouri** [comment]
I've made `showMoreResults` a query param.  Should work well with back button now.
This may be worth landing and then addressing the react-virtualized issues separately.

If this become a blocker it would be trivial so render the entire list instead.

**Comment by samouri** [comment]
just had a pretty tricky rebase....hopefully its all good!

**Comment by samouri** [comment]
I think i'd like to land this as-is.  There are a few items to address in followup prs:
1. loading states
2. what to do when search results have less then 4 (or < 200 for expanded) results
3. rv bug (and maybe correlated clipping. are styles all placed in the right spots?)


#### To Test
1. go to: https://calypso.live/following/manage?branch=add/reader/som-show-more-results
2. verify behavior is as described https://github.com/Automattic/wp-calypso/pull/13480#issue-224917780

**Comment by jancavan** [comment]
The "Show More" button somehow changed a bit. Updated it in: https://github.com/Automattic/wp-calypso/pull/13480/commits/bff0197a6f443df79ad8392c3b95fa6a9999f22d

**Comment by jancavan** [comment]
I noticed that the site you just followed gets added to your Following list, but shows an empty block.

Steps to reproduce:
1. Type in a keyword in Search
2. Follow one of the sites in the list
3. Notice that it gets added, but is empty

Video: https://cloudup.com/c7RZHCEtgG8

**Comment by jancavan** [comment]
Also I think based on the original mockup, it should show a site count:

screenshot 2017-05-02 13 27 44

Is this for a separate PR?

**Comment by samouri** [comment]
> Also I think based on the original mockup, it should show a site count:


@jancavan, we don't actually get that number in the api response yet. I'm working now to see if we can.  What should the copy be for 200+ (but we only page to 200).

**Comment by blowery** [comment]
looking at this 👀

**Comment by blowery** [comment]
seems like we're using page.replace instead of page.view, so no history entries are being created. That means that the back button ends up leaving the page entirely... Is that what's expected currently?

Otherwise works pretty well!

File: client/reader/following-manage/feed-search-results.jsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		);
+		return (
+			<div className="following-manage__search-results">
+				{ resultsToShow }
+				<div className="following-manage__show-more">
+					<Button compact icon
+						onClick={ showMoreResultsClicked }
+						className="following-manage__show-more-button button">
+							<Gridicon icon="chevron-down" />
+							Show more
```
Comment (blowery) [inline client/reader/following-manage/feed-search-results.jsx]: needs translating

**Comment by jancavan** [comment]
> Also I think based on the original mockup, it should show a site count

As discussed, scratch this. Button shouldn't display the site count since this button disappears anyway after clicking and the site results just take over the entire page.

**Comment by samouri** [comment]
@blowery, lets talk offline about the `page` behavior.  I'm not sure whats expected there and would be happy to swap to page.view

**Pull request merged**